src/components/Microsoft.DotNet.UpgradeAssistant.Extensions/ExtensionAssemblyLoadContext.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions
+{
+    internal class ExtensionAssemblyLoadContext : AssemblyLoadContext
+    {
+        private const string ALC_PREFIX = "UA_";
+
+        private readonly ExtensionInstance _extension;
+
+        public ExtensionAssemblyLoadContext(ExtensionInstance extension)
+            : base(ALC_PREFIX + extension.Name)
```

Comment:
Interesting, were you hitting name conflicts here?

Reply:
The name of our default extension is Default and the default alc is Default